core/src/main/java/io/kestra/core/runners/RunContext.java

```diff
@@ -583,6 +593,19 @@ public RunContext forWorkingDirectory(ApplicationContext applicationContext, Wor
         return newContext;
     }
 
+    public RunContext forWorkingDirectoryTask() {
+        Map<String, Object> decryptedVariables = new HashMap<>();
```

Comment:
You can directly init the decryptedVariables map with  this.variables, and only update the `outputs` entry instead of iterating over all entries.